src/target/riscv/riscv.c

```diff
@@ -28,6 +28,14 @@
 
 #define get_field(reg, mask) (((reg) & (mask)) / ((mask) & ~((mask) << 1)))
 #define set_field(reg, mask, val) (((reg) & ~(mask)) | (((val) * ((mask) & ~((mask) << 1))) & (mask)))
+#define set_2_fields(reg, m1, v1, ...) set_field(set_field(reg, m1, v1), __VA_ARGS__)
+#define set_3_fields(reg, m1, v1, ...) set_2_fields(set_field(reg, m1, v1), __VA_ARGS__)
+#define set_4_fields(reg, m1, v1, ...) set_3_fields(set_field(reg, m1, v1), __VA_ARGS__)
+#define set_5_fields(reg, m1, v1, ...) set_4_fields(set_field(reg, m1, v1), __VA_ARGS__)
+#define set_6_fields(reg, m1, v1, ...) set_5_fields(set_field(reg, m1, v1), __VA_ARGS__)
+#define set_7_fields(reg, m1, v1, ...) set_6_fields(set_field(reg, m1, v1), __VA_ARGS__)
+#define set_8_fields(reg, m1, v1, ...) set_7_fields(set_field(reg, m1, v1), __VA_ARGS__)
+#define set_9_fields(reg, m1, v1, ...) set_8_fields(set_field(reg, m1, v1), __VA_ARGS__)
```

Comment:
How about:
```
#define field_value(mask, val) set_field(0, mask, val)
```

Then use that everywhere with | in between, instead of the set_N_fields macros. In the places where you use them you still don't care about the initial value, and it seems awkward to count the number of arguments and counting all the commas in a macro with 19 arguments.

It's fine to acknowledge that these macros are setting bit fields. We don't have to hide the fact that these can be combined with bit-wise OR inside a macro, because that will never change.

Reply:
Addressed.